File: dia-02/08-trabalhando-com-strings/03-exercicio1-corrigido.cs
// 1. Exploração de strings

using System;

class App
{
    static void Main()
    {
        Console.Write("Digite seu nome: ");
        string nome = Console.ReadLine();

        Console.Write("Digite seu sobrenome: ");
        string sobrenome = Console.ReadLine();

        Console.WriteLine($"{nome} {sobrenome}");

        string inicialNome = nome.Substring(0, 1).ToUpper();
        string inicialSobrenome = sobrenome.Substring(0, 1).ToUpper();
        int totalCaracteres = nome.Length + sobrenome.Length;
        Console.WriteLine($"Iniciais e contagem: {inicialNome}.{inicialSobrenome}. ({totalCaracteres})");

        int posicaoMeioNome = nome.Length / 2;
        string primeiraMetadeNome = nome.Substring(0, posicaoMeioNome);
        int posicaoMeioSobrenome = sobrenome.Length / 2;
        string segundaMetadeSobrenome = sobrenome.Substring(posicaoMeioSobrenome);
        string nomeSecreto = $"{primeiraMetadeNome}{segundaMetadeSobrenome}"; // ou string.Concat(primeiraMetadeNome, segundaMetadeSobrenome);
        Console.WriteLine($"Nome secreto: {nomeSecreto}");
    }
}

File: MM5 Front/curso-adaptacao-js-edicao-2025/dia-01/05-strings/03-exercicio1-corrigido.js
let nome = "john";
let sobrenome = "doe";

console.log(`Nome completo: ${nome} ${sobrenome}`);

let inicialNome = nome.charAt(0).toUpperCase();
let inicialSobrenome = sobrenome.charAt(0).toUpperCase();

let contagemTotal = nome.length + sobrenome.length;

console.log("Iniciais e contagem: " + inicialNome + "." + inicialSobrenome + " (" + contagemTotal + ")");

// Criação do nome secreto
const pontoCorteNome = Math.ceil(nome.length / 2); 
const primeiraMetadeNome = nome.slice(0, pontoCorteNome);

const pontoCorteSobrenome = Math.floor(sobrenome.length / 2);
const segundaMetadeSobrenome = sobrenome.slice(pontoCorteSobrenome); 

const nomeSecreto = primeiraMetadeNome + segundaMetadeSobrenome;

console.log(`Nome secreto: ${nomeSecreto}`); 
